src/planet-a/SubdivTriangleMesh.h

```diff
@@ -0,0 +1,487 @@
+     */
+    constexpr uint32_t vertex_offset_shared() const noexcept { return m_vrtxSharedOffset; }
+
+    constexpr SharedVrtxId vertex_to_shared(VertexId sharedId) const noexcept { return SharedVrtxId(uint32_t(sharedId) - m_vrtxSharedOffset); }
+
+    constexpr uint32_t vertex_offset_fill(ChunkId chunkId) const noexcept
+    {
+        return m_chunkVrtxFillCount * uint32_t(chunkId);
+    }
+
+    constexpr bool vertex_is_shared(VertexId vrtx) const noexcept
+    {
+        return uint32_t(vrtx) >= m_vrtxSharedOffset;
+    }
+
+    void clear(SubdivTriangleSkeleton& rSkel)
+    {
+        // Delete all chunks
+        m_chunkIds.for_each([this, &rSkel] (ChunkId chunkId)
+        {
+            // Release their associated skeleton triangle
+            rSkel.tri_release(m_chunkData[size_t(chunkId)].m_skeletonTri);
+
+            // Release all of their shared vertices
+            for (SharedVrtxStorage_t& shared : chunk_shared_mutable(chunkId))
+            {
+                shared_release(shared);
+            }
+        });
+
+        // Delete all shared vertices
+        m_sharedIds.for_each([this, &rSkel] (SharedVrtxId sharedId)
+        {
+           // Release associated skeleton vertex
+           rSkel.vrtx_release(m_sharedSkVrtx[size_t(sharedId)]);
+        });
+    }
+
+private:
+
+
+    ArrayView_t<SharedVrtxStorage_t> chunk_shared_mutable(ChunkId chunkId) noexcept
+    {
+        size_t const offset = size_t(chunkId) * m_chunkVrtxSharedCount;
+        return {&m_chunkSharedUsed[offset], m_chunkVrtxSharedCount};
+    }
+
+    SharedVrtxId chunk_shared_get(ChunkId chunkId, ChunkLocalSharedId localId) const noexcept
+    {
+        return m_chunkSharedUsed[size_t(chunkId) * m_chunkVrtxSharedCount
+                                 + size_t(localId)];
+    }
+
+    /**
+     * @brief Create or get a shared vertex associated with a skeleton vertex
+     *
+     * @param skVrtxId
+     * @return
+     */
+    SharedVrtxId shared_get_or_create(SkVrtxId skVrtxId,
+                                      SubdivTriangleSkeleton &rSkel)
+    {
+        auto const& [it, success]
+                = m_skVrtxToShared.try_emplace(skVrtxId, SharedVrtxId(0));
+        if (success)
+        {
+            SharedVrtxId const id = m_sharedIds.create();
+            it->second = id;
+            m_sharedFaceCount[size_t(id)] = 0;
+            m_sharedSkVrtx[size_t(id)] = rSkel.vrtx_store(skVrtxId);
+            m_sharedNewlyAdded.push_back(id);
+        }
+
+        return it->second;
+    }
+
+    SharedVrtxStorage_t shared_store(SharedVrtxId triId)
+    {
+        return m_sharedRefCount.ref_add(triId);
+    }
+
+    void shared_release(SharedVrtxStorage_t &rStorage) noexcept
+    {
+        m_sharedRefCount.ref_release(rStorage);
+    }
+
+    constexpr VertexId shared_get_vrtx(SharedVrtxId sharedId) const
+    {
+        return VertexId(uint32_t(m_vrtxSharedOffset) + uint32_t(sharedId));
+    };
+
+    uint16_t m_chunkMax;
```

Comment:
Consider re-arranging the member variables here for optimial struct packing.

Reply:
Unfortunately that would mess with the initialization order. Perhaps the member initializers in the constructor was a bad idea?